src/de/willuhn/jameica/hbci/gui/parts/AbstractSepaSammelTransferList.java

```diff
@@ -116,8 +116,7 @@ public void format(TableItem item) {
             item.setForeground(Color.COMMENT.getSWTColor());
 
           // Checken, ob der Auftrag einen Reminder hat oder ob es ein geclonter Auftrag ist
-          HibiscusDBObject o = (HibiscusDBObject) l;
```

Comment:
Durch Entfernen der lokalen Variable `o`, wird die Semantik der Identität bei ihrer Verwendung gebrochen. Der Code sollte daher an dieser Stelle bleiben, wie er war.

Reply:
Hallo @meigelb,
ehrlich gesagt, verstehe ich nicht, was Du mit "die Semantik der Identität bei ihrer Verwendung" ausdrücken möchtest.

Die Aussage ist "checke ob `l` ein Klon ist", wozu in Zeile 119 die UUID und in Zeile 133 das Template verwendet wird.

Nach meinem Verständnis ist hier nicht einmal der Downcast notwendig, da die Variable `l` vom Typ `SepaSammelTransfer` ist, was wiederum ist ein `HibiscusDBObject` ist:
```java
public interface SepaSammelTransfer<T extends SepaSammelTransferBuchung> extends HibiscusDBObject, Terminable, Duplicatable, SepaPayment
```

Reply:
Hm, ich weiß auch nicht, was ich damals dort gesehen habe. Aus heutiger Sicht passt es. Vielleicht habe ich nicht erkannt, dass `l` ein Objekt vom Typ `SepaSammelTransfer` ist, sondern das als eine `1` gelesen. Danke fürs kritische Hinterfragen.